src/gt4py/next/embedded/common.py

```diff
@@ -146,3 +146,31 @@ def _find_index_of_dim(
         if dim == d:
             return i
     return None
+
+
+def canonicalize_any_index_sequence(
+    index: common.AnyIndexSpec,
+) -> common.AnyIndexSpec:
+    new_index: common.AnyIndexSpec = (index,) if isinstance(index, slice) else index
```

Comment:
```suggestion
    # TODO: instead of canonicalizing to `NamedRange`, we should canonicalize to `NamedSlice`
    new_index: common.AnyIndexSpec = (index,) if isinstance(index, slice) else index
```